src/devices/LidarLiteV3/LidarLiteV3.cs

```diff
@@ -0,0 +1,358 @@
+            _powerEnablePin = powerEnablePin;
+
+            if (_powerEnablePin != -1)
+            {
+                _gpioController.OpenPin(_powerEnablePin, PinMode.Output);
+                PowerOn();
+            }
+
+            Reset();
+        }
+
+        /// <summary>
+        /// Power off the device if GPIO controller and power enable pin is provided.
+        /// </summary>
+        public void PowerOff()
+        {
+            if (_powerEnablePin != -1)
+            {
+                _gpioController.Write(_powerEnablePin, PinValue.Low);
+            }
+            else
+            {
+                throw new InvalidOperationException("Cannot power off without providing GPIO controller and power enable pin.");
+            }
+        }
+
+        /// <summary>
+        /// Power on the device if GPIO controller and power enable pin is provided.
+        /// </summary>
+        public void PowerOn()
+        {
+            if (_powerEnablePin != -1)
+            {
+                _gpioController.Write(_powerEnablePin, PinValue.High);
+            }
+            else
+            {
+                throw new InvalidOperationException("Cannot power off without providing GPIO controller and power enable pin.");
+            }
+        }
+
+        /// <summary>
+        /// Reset FPGA, all registers return to default values
+        /// </summary>
+        public void Reset()
+        {
+            WriteRegister(Register.ACQ_COMMAND, 0x00);
+        }
+
+        /// <summary>
+        /// Measure distance in cm
+        /// </summary>
+        /// <remarks>
+        /// Note: Do not call if running while in repetition mode.  It will block until
+        /// repetition finishes (forever if infinite).
+        /// </remarks>
+        /// <param name="withReceiverBiasCorrection">Faster without bias correction, but more prone to errors if condition changes.</param>
+        /// <returns>Distance in cm</returns>
+        public async Task<int> MeasureDistanceAsync(bool withReceiverBiasCorrection = true)
+        {
+            WriteRegister(Register.ACQ_COMMAND, withReceiverBiasCorrection ? (byte)0x04 : (byte)0x03);
+
+            while (Status.HasFlag(SystemStatus.BusyFlag))
+            {
+                await Task.Delay(4);
+            }
+
+            return LastDistance;
+        }
+
+        /// <summary>
+        /// Measure distance in cm
+        /// </summary>
+        /// <remarks>
+        /// Note: Do not call if running while in repetition mode.  It will block until
+        /// repetition finishes (forever if infinite).
+        /// </remarks>
+        /// <param name="withReceiverBiasCorrection">Faster without bias correction, but more prone to errors if condition changes.</param>
+        /// <returns>Distance in cm</returns>
+        public int MeasureDistance(bool withReceiverBiasCorrection = true)
+        {
+            return MeasureDistanceAsync(withReceiverBiasCorrection).Result;
```

Comment:
Instead of calling .Result, you should be calling `.GetAwaiter().GetResult()` or else you could run into deadlocks when the task is not finished.

Reply:
Ended up getting rid of the async method because DelayHelper wasn't async and that replaced the need for async.